LoopKit/DeviceAlert.swift

```diff
@@ -43,45 +43,48 @@ public struct UserAlert {
         public let isCritical: Bool
         // TODO: when we have more complicated actions.  For now, all we have is "acknowledge".
 //        let actions: [UserAlertAction]
-        public let acknowledgeAction: String
-        public init(title: String, body: String, acknowledgeAction: String, isCritical: Bool = false) {
+        public let acknowledgeActionButtonLabel: String
+        public init(title: String, body: String, acknowledgeActionButtonLabel: String, isCritical: Bool = false) {
             self.title = title
             self.body = body
-            self.acknowledgeAction = acknowledgeAction
+            self.acknowledgeActionButtonLabel = acknowledgeActionButtonLabel
             self.isCritical = isCritical
         }
     }
+    public struct Identifier: Equatable {
+        /// Unique device manager identifier from whence the alert came, and to which alert acknowledgements should be directed.
+        public let deviceManagerInstanceIdentifier: String
+        /// Per-alert unique identifier, for instance to group alert types.  This is the identifier that will be used to acknowledge the alert.
+        public let typeIdentifier: TypeIdentifier
+        public init(deviceManagerInstanceIdentifier: String, typeIdentifier: TypeIdentifier) {
+            self.deviceManagerInstanceIdentifier = deviceManagerInstanceIdentifier
+            self.typeIdentifier = typeIdentifier
+        }
+        /// An opaque value for this tuple for unique identification of the alert across devices.
+        public var value: String {
+            return "\(deviceManagerInstanceIdentifier).\(typeIdentifier)"
+        }
+    }
```

Comment:
@nhamming I discussed the name of this variable at length with @ps2 .  This isn't a `DeviceManager`, so `deviceManager` won't work.  I tried `deviceIdentifier` but that name is already taken for other things.  Since Pete and I agreed it is really the identifier for the "instance", hence the (somewhat wordy) `deviceManagerInstanceIdentifier`.  I would be fine with `deviceManagerIdentifier`, though.  WDYT?

Reply:
Here my comment is the `Identifier` part is unneeded, since the struct is `Identifier` (not add on to my comment about using `Instance`). So, it would be `Identifier.deviceManager` instead of Identifier.deviceManagerIdentifier`. But I'm not holding strong to this comment.